shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -46,6 +46,10 @@
   // Current coordinate of the mouse cursor in physical device pixels.
   CGPoint location = CGPointZero;
 
+  // True if flutter::PointerData::Change::kAdd has been sent to Flutter for the discrete scrolling
+  // pointer.
+  BOOL discrete_scroll_pointer_added = NO;
```

Comment:
> I realized if we only send discrete scroll events on a pointer, it doesn't need to be removed, since it won't ever trigger any hover events. So the code is simpler now, no need to coordinate the reuse of the hover pointer. 

Not sure if I follow this line. Could you explain more why this logic is now only for discrete scroll?

Reply:
When we send Hover event, we need to be careful to remove that pointer identifier when the mouse or trackpad is disconnected, or else the hover will continue in the same location. This is a concern on iPad since the keyboard case is could be removed at any time. But if a pointer identifier never has any Hover event, only Scroll event, we can leave it around without removing it. There won't be any side effects from keeping it around.

Reply:
> But if a pointer identifier never has any Hover event, only Scroll event, we can leave it around without removing it.

This is referring to mouse wheel? How is it related to this `discrete_scroll_pointer_added` flag?

Reply:
discrete scroll = mouse wheel scroll. We need `discrete_scroll_pointer_added` since it is required to send an `Add` event before sending any `Scroll` events. That flag will make sure the event is only sent once, the first time a `Scroll` occurs.

Reply:
I understand how `discrete_scroll_pointer_added` is used. I am not sure how it is related to the hover event and why we don't need this flag for continuous scroll.

Reply:
If you send an event of phase `kPanZoomStart` (which is used for continuous scroll and pinch), the embedder will synthesize the add event for you (see `pointer_data_packet_converter.cc`). It doesn't do that if you send an event of event type `kScroll`.

Hover pointer is now separate from discrete scroll pointer. In the previous version where they used the same identifier, there could be an assert failure in the rare case that scroll occurred before mouse moving, since `Scroll` would occur before `Add`. Now `Add` will be sent for scroll based on `discrete_scroll_pointer_added`, `Add` will be sent for hover based on the phase of `UIHoverGestureRecognizer`, and `Add` will be added automatically elsewhere in the embedder for the continuous scroll and pinch events.

Reply:
Can you add this to the comment? just from the code It's hard to tell why you only have this flag for discrete but not for continuous case. 

Also is it possible to handle `add` event for discrete scroll in `pointer_data_packet_converter.cc` to be consistent with other gestures?